packages/taco/integration-test/encrypt-decrypt.test.ts

```diff
@@ -0,0 +1,140 @@
+import { beforeAll, describe, expect, test } from 'vitest';
+
+import { domains, fromBytes, toBytes } from '@nucypher/shared';
+import {
+  EIP4361AuthProvider,
+  USER_ADDRESS_PARAM_DEFAULT,
+} from '@nucypher/taco-auth';
+import * as dotenv from 'dotenv';
+import { ethers } from 'ethers';
+import {
+  conditions,
+  decrypt,
+  encrypt,
+  initialize,
+  ThresholdMessageKit,
+} from '../src';
+
+dotenv.config({ path: 'integration-test/.env' });
+
+// TODO: Enable and configure evn variables for this test as part of https://github.com/nucypher/taco-web/issues/638
+// To run the test locally, unskip and use env variables similar to integration-test/.env.example
+describe.skip('Taco Encrypt/Decrypt Integration Test', () => {
+  let provider: ethers.providers.JsonRpcProvider;
+  let domain: string;
+  let ritualId: number;
+  let chainId: number;
+  let encryptorSigner: ethers.Wallet;
+  let consumerSigner: ethers.Wallet;
+
+  beforeAll(async () => {
+    const rpcProviderUrl = process.env.RPC_PROVIDER_URL;
+    if (!rpcProviderUrl) {
+      throw new Error('RPC_PROVIDER_URL is not set.');
+    }
+
+    const encryptorPrivateKey = process.env.ENCRYPTOR_PRIVATE_KEY;
+    if (!encryptorPrivateKey) {
+      throw new Error('ENCRYPTOR_PRIVATE_KEY is not set.');
+    }
+
+    const consumerPrivateKey = process.env.CONSUMER_PRIVATE_KEY;
+    if (!consumerPrivateKey) {
+      throw new Error('CONSUMER_PRIVATE_KEY is not set.');
+    }
+
+    domain = process.env.DOMAIN || domains.TESTNET;
+    ritualId = parseInt(process.env.RITUAL_ID || '6');
+    provider = new ethers.providers.JsonRpcProvider(rpcProviderUrl);
+
+    const CHAIN_ID_FOR_DOMAIN = {
+      [domains.MAINNET]: 137,
+      [domains.TESTNET]: 80002,
+      [domains.DEVNET]: 80002,
+    };
+    chainId = CHAIN_ID_FOR_DOMAIN[domain];
+
+    encryptorSigner = new ethers.Wallet(encryptorPrivateKey, provider);
+    consumerSigner = new ethers.Wallet(consumerPrivateKey, provider);
+
+    // Initialize the library
+    await initialize();
+
+    // Verify network connection
+    const network = await provider.getNetwork();
+    if (network.chainId !== chainId) {
+      throw new Error(
+        `Provider connected to wrong network. Expected ${chainId}, got ${network.chainId}`,
+      );
+    }
+  });
+
+  test.each([
+    0,
+    1000000000000000000000, // 1000 ETH/PLY
+    BigInt(10000000000000000), // 0.01 ETH/PLY
+  ])(
+    'should encrypt and decrypt a message with condition value %s',
+    async (value) => {
+      // Create test message
+      const messageString = 'This is a secret 🤐';
+      const message = toBytes(messageString);
+
+      // Create condition
+      const hasPositiveBalance = new conditions.base.rpc.RpcCondition({
```

Comment:
I think you can just do a single `CompoundCondition` (`AND`) comprised of multiple rpc conditions each using the different values. Then you can just run this once.

Also I would say once of those should be a check for a balance < uint256 max i.e.
```typescript
returnValueTest: {
    comparator: "<",
    value: BigInt("115792089237316195423570985008687907853269984665640564039457584007913129639935"),
}
```

Reply:
The suggestion on using `CompoundCondition` I think it is great for the case.
And yes, it would be overkill to repeat the full test just for different condition values. I basically made it like this to make it easy for this to be checked with the nodes as part of the issue scope definition.

Probably for an integration test it is good to use extensive complicated conditions that covers almost all conditions. Would be good to craft such when doing https://github.com/nucypher/taco-web/issues/638.